src/nn/mod.rs

```diff
@@ -147,7 +147,7 @@ impl Network {
 pub fn sigmoid(arr: &DVector<f32>) -> DVector<f32> {
     let mut sig = arr.clone();
     for elem in sig.iter_mut() {
-        *elem = 1.0 / (1.0 + (elem).exp());
+        *elem = 1.0 / (1.0 + (-1.0f32 * elem.clone()).exp());
```

Comment:
I think it's slightly preferred to dereference `Copy` types instead of cloning them. A `clone()` always looks like: expensive. So maybe rather: `(-1.0f32 * *elem)`  (and I know, those two `*` look kinda awkward)